### PR TITLE
Post installation verifications for FaaS

### PR DESCRIPTION
**Pull request by jilju**
Creted new functions for FaaS provider and consumer verification.
Movied the verification of pods in managed-fusion namespace to the new function.
Verify security in ocs-client-operator pod on consumer.
Verifiy storage client on consumer.

Consumer:
    1. Verify CSV phase
    2. Verify client endpoint

Provider:
    1. Verify CSV phase
    2. Verify ocs-provider-server pod is Running
    4. Verify that Cephcluster is Ready and hostNetworking is True
    5. Verify that the security groups are set up correctly

### CODE REVIEW

File: ocs_ci/helpers/managed_services.py [hunk 1]
```diff
@@ -363,3 +370,235 @@ def verify_pods_in_managed_fusion_namespace():
...
+    # Verify attributes specific to cluster types
+    if config.ENV_DATA["cluster_type"].lower() == "provider":
+        sc_obj = OCP(
+            kind=constants.STORAGECLUSTER,
+            namespace=config.ENV_DATA["cluster_namespace"],
+        )
+        sc_data = sc_obj.get()["items"][0]
+        verify_faas_provider_storagecluster(sc_data)
+        verify_faas_provider_resources()
+        if get_ocs_osd_deployer_version() >= get_semantic_version("2.0.11-0"):
```
Comment (yitzhak12) [inline ocs_ci/helpers/managed_services.py:395]: This function `get_ocs_osd_deployer_version` should be updated for the `managed-fusion` namespace. (e.g., change the namespace to `config.ENV_DATA["cluster_namespace"]` and the constant `constants.OCS_OSD_DEPLOYER_CSV_LABEL` should be replaced.

File: ocs_ci/helpers/managed_services.py [hunk 2]
```diff
@@ -363,3 +370,235 @@ def verify_pods_in_managed_fusion_namespace():
...
+            verify_provider_topology()
```
Comment (yitzhak12) [inline ocs_ci/helpers/managed_services.py:396]: I think this function should also be updated for the `managed-fusion` namespace.

**Review by yitzhak12 — COMMENTED** [review]
The pr looks good

File: ocs_ci/helpers/managed_services.py (hunk 2 above)
Comment (jilju) [reply ocs_ci/helpers/managed_services.py]: The function is already updated tin master to use the correct namespace.

**Comment by jilju** [comment]
Added other updates to unblock testing of the PR. 
1. Added checks for machineconfigpool before and after udating the pull secret.
2. Increased wit time for nodes after creating rosa cluster. The wait time is based on observations.

**Comment by jilju** [comment]
Verified:
Provider - https://ocs4-jenkins-csb-odf-qe.apps.ocp-c1.prod.psi.redhat.com/job/qe-deploy-ocs-cluster/24229
Consumer - https://ocs4-jenkins-csb-odf-qe.apps.ocp-c1.prod.psi.redhat.com/job/qe-deploy-ocs-cluster/24228

**Review by yitzhak12 — APPROVED** [review]

**Pull request merged**